flutter_appauth/ios/Classes/FlutterAppAuth.m

```diff
@@ -37,8 +37,49 @@ + (NSMutableDictionary *)processResponses:(OIDTokenResponse*) tokenResponse auth
     return processedResponses;
 }
 
-+ (void)finishWithError:(NSString *)errorCode message:(NSString *)message  result:(FlutterResult)result {
-    result([FlutterError errorWithCode:errorCode message:message details:nil]);
++ (void)finishWithError:(NSString *)errorCode message:(NSString *)message result:(FlutterResult)result error:(NSError * _Nullable)error {
+
+    NSMutableDictionary<NSString *, id> *details = [NSMutableDictionary dictionary];
+
+    if (error) {
+        id authError = error.userInfo[OIDOAuthErrorResponseErrorKey];
+        NSDictionary<NSString *, id> *authErrorMap = [authError isKindOfClass:[NSDictionary class]] ? authError : nil;
+        
+        if (authErrorMap) {
+            if ([authErrorMap objectForKey:@"error"]) {
+                [details setObject:authErrorMap[@"error"] forKey:@"error"];
+            }
+            if ([authErrorMap objectForKey:@"error_description"]) {
+                [details setObject:authErrorMap[@"error_description"] forKey:@"error_description"];
+            }
+        }
+        if (error.domain) {
+            [details setObject:error.domain forKey:@"type"];
+        }
+        if (error.code) {
+            [details setObject:[@(error.code) stringValue] forKey:@"code"];
+        }
+        
+        id underlyingErr = [error.userInfo objectForKey:NSUnderlyingErrorKey];
+        NSError *underlyingError = [underlyingErr isKindOfClass:[NSError class]] ? underlyingErr : nil;
+        if (underlyingError) {
+            if (underlyingError.domain) {
+                [details setObject:underlyingError.domain forKey:@"error_uri"];
```

Comment:
If there's an intent to map the `domain` then unless I'm missing something, this should go to a separate property instead of here. The reason I mention this is that OAuth specs are specific on what  `error_uri` means (https://datatracker.ietf.org/doc/html/rfc6749#section-5.2). The SDKs extract this from reading the `error_uri` query string parameter of the redirect response.  On Android, this goes to a dedicated property so is handled more cleanly. From what I could tell, the AppAuth iOS SDK doesn't do so nor does it set it as the domain. [This code](https://github.com/openid/AppAuth-iOS/blob/c89ed571ae140f8eb1142735e6e23d7bb8c34cb2/Sources/AppAuthCore/OIDErrorUtilities.m#L104) indicates it's part of the description in the `NSError`'s `userInfo` dicationary

Based on this, if `errorUri` will be kept on the Dart side then it would be more accurate to tell devs that this is only populate on Android and that extracting it from iOS needs to be done differently. The latter may be difficult to do so I'm not expecting it to be covered in the PR. I believe for this to be done more easily, the AppAuth iOS SDK would need another key to save the value of the `error_uri` against the `userInfo`

Reply:
Ah yes you're right, I think I just assumed that was directly mapped 🤦 . I'll expose the domain an iOS as a separate property just so it's not missed and only map the error uri on Android